src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -213,68 +199,54 @@ public IEnumerable<ManifestVersionUpdate> CalculateManifestRollbacks(string roll
             if (unrecognizedManifestIds.Any())
             {
                 _reporter.WriteLine(string.Format(LocalizableStrings.RollbackDefinitionContainsExtraneousManifestIds, rollbackDefinitionFilePath, string.Join(" ", unrecognizedManifestIds)).Yellow());
-                manifestRollbacks = manifestRollbacks.Where(rollbackManifest => currentManifestIds.Contains(rollbackManifest.Item1));
+                manifestRollbacks = manifestRollbacks.Where(rollbackManifest => currentManifestIds.Contains(rollbackManifest.Id));
             }
 
-            var manifestUpdates = manifestRollbacks
-                .Select(manifest =>
-                {
-                    var installedManifestInfo = GetInstalledManifestVersion(manifest.id);
-                    return new ManifestVersionUpdate(manifest.id, installedManifestInfo.manifestVersion, installedManifestInfo.sdkFeatureBand.ToString(),
-                        manifest.version, manifest.featureBand.ToString());
-                });
+            var manifestUpdates = manifestRollbacks.Select(manifest =>
+            {
+                var (id, (version, band)) = manifest;
+                var (installedVersion, installedBand) = GetInstalledManifestVersion(id);
+                return new ManifestVersionUpdate(id, installedVersion, installedBand.ToString(), version, band.ToString());
+            });
 
             return manifestUpdates;
         }
 
-
         public async Task<IEnumerable<WorkloadDownload>> GetManifestPackageDownloadsAsync(bool includePreviews, SdkFeatureBand providedSdkFeatureBand, SdkFeatureBand installedSdkFeatureBand)
         {
             var downloads = new List<WorkloadDownload>();
             foreach (var manifest in _workloadResolver.GetInstalledManifests())
             {
                 try
                 {
-                    var packageId = _workloadManifestInstaller.GetManifestPackageId(new ManifestId(manifest.Id), providedSdkFeatureBand);
-
-                    bool success;
-                    // After checking the --sdk-version, check the current sdk band, and then the manifest band in that order
-                    (success, var latestVersion) = await GetPackageVersion(packageId, packageSourceLocation: _packageSourceLocation, includePreview: includePreviews);
-                    if (success)
-                    {
-                        downloads.Add(new WorkloadDownload(manifest.Id, packageId.ToString(), latestVersion.ToString()));
-                    }
-
-                    if (!success && !installedSdkFeatureBand.Equals(providedSdkFeatureBand))
+                    PackageId? providedPackageId = null;
+                    var fallbackFeatureBand = new SdkFeatureBand(manifest.ManifestFeatureBand);
+                    SdkFeatureBand[] bands = [providedSdkFeatureBand, installedSdkFeatureBand, fallbackFeatureBand];
```

Comment:
I wasn't actually familiar with this notation. Nice.

Reply:
Collection expressions are a new thing. But they do come in handy and are really clean compared to traditional collection initializers.